reinforcement_learning/unit_test/live_model_test.cc

```diff
@@ -231,3 +232,87 @@ BOOST_AUTO_TEST_CASE(live_model_mocks) {
+  });
+  Fake(Dtor(mock_observation_logger));
+
+  Mock<r::i_logger> mock_interaction_logger;
+  std::vector<std::string> recorded_interactions;
+  When(Method(mock_interaction_logger, init)).AlwaysReturn(err::success);
+  When(Method(mock_interaction_logger, append)).AlwaysDo(
+    [&recorded_interactions](std::string& message, reinforcement_learning::api_status* status) {
+    recorded_interactions.push_back(message); return err::success;
+  });
+  Fake(Dtor(mock_interaction_logger));
+
+  Mock<m::i_data_transport> mock_data_transport;
+  When(Method(mock_data_transport, get_data)).AlwaysReturn(err::success);
+  Fake(Dtor(mock_data_transport));
+
+  Mock<m::i_model> mock_model;
+  When(Method(mock_model, update)).AlwaysReturn(err::success);
+  When(Method(mock_model, choose_rank)).AlwaysReturn(err::success);
+  Fake(Dtor(mock_model));
+
+  r::logger_factory_t logger_factory;
+  logger_factory.register_type(r::value::OBSERVATION_EH_LOGGER,
+    [&mock_observation_logger](r::i_logger** retval, const u::configuration&, r::error_callback_fn*, r::api_status*) { *retval = &mock_observation_logger.get(); return err::success; });
+  logger_factory.register_type(r::value::INTERACTION_EH_LOGGER,
+    [&mock_interaction_logger](r::i_logger** retval, const u::configuration&, r::error_callback_fn*, r::api_status*) { *retval = &mock_interaction_logger.get(); return err::success; });
+
+  r::data_transport_factory_t data_transport_factory;
+  data_transport_factory.register_type(r::value::AZURE_STORAGE_BLOB,
+    [&mock_data_transport](m::i_data_transport** retval, const u::configuration&, r::api_status*) {*retval = &mock_data_transport.get(); return err::success; });
+  r::model_factory_t model_factory;
+  model_factory.register_type(r::value::VW,
+    [&mock_model](m::i_model** retval, const u::configuration&, r::api_status*) {*retval = &mock_model.get(); return err::success; });
+
+  u::configuration config;
+  cfg::create_from_json(JSON_CFG, config);
+  config.set(r::name::EH_TEST, "true");
+
+  r::live_model model(config, nullptr, nullptr, &data_transport_factory, &model_factory, &logger_factory);
+
+  r::api_status status;
+  BOOST_CHECK_EQUAL(model.init(&status), err::success);
+
+  auto const event_id_1 = "event_id";
+  auto const event_id_2 = "event_id_2";
+
+  r::ranking_response response;
+  auto const num_iterations = 5;
+  for (auto i = 0; i < num_iterations; i++) {
+    BOOST_CHECK_EQUAL(model.choose_rank(event_id_1, JSON_CONTEXT, response), err::success);
+    BOOST_CHECK_EQUAL(model.report_outcome(event_id_1, 1.0), err::success);
+
+    BOOST_CHECK_EQUAL(model.choose_rank(event_id_2, JSON_CONTEXT, response), err::success);
+    BOOST_CHECK_EQUAL(model.report_outcome(event_id_2, 1.0), err::success);
+  }
+
+  auto const expected_interaction_1 = u::concat(R"({"Version":"1","EventId":")", event_id_1, R"(","a":[1,2],"c":)", JSON_CONTEXT, R"(,"p":[0.500000,0.500000],"VWState":{"m":"N/A"}})");
```

Comment:
>1 [](start = 63, length = 1)

I see "Version" is hard coded in ranking_event.cc but this should probably be dynamic. As the version increases the tests should hold.

Reply:
Version is currently also hardcoded in the event serialization [see here](https://github.com/JohnLangford/vowpal_wabbit/blob/master/reinforcement_learning/rlclientlib/ranking_event.cc#L14). I agree we want to make it clear and easy to change this when the version does change.